### PR TITLE
[OSLog] Add bindings for xcode11

### PR DESCRIPTION
**Pull request by wachaudh**
https://github.com/xamarin/xamarin-macios/wiki/OSLog-macOS-xcode11-beta5

### CODE REVIEW

File: src/oslog.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+		[Export ("entriesEnumeratorAndReturnError:")]
+		[return: NullAllowed]
+		NSEnumerator EntriesEnumeratorAndReturnError ([NullAllowed] out NSError error);
+
+		[Export ("positionWithDate:")]
+		NSObject PositionWithDate (NSDate date);
+
+		[Export ("positionWithTimeIntervalSinceEnd:")]
+		NSObject PositionWithTimeIntervalSinceEnd (double seconds);
```
Comment (rolfbjarne) [inline src/oslog.cs]: Method must contain a verb:

```suggestion
		NSObject GetPositionWithTimeIntervalSinceEnd (double seconds);
```

And shouldn't the return type be OSLogPosition?

File: src/oslog.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+		[Export ("positionWithTimeIntervalSinceLatestBoot:")]
+		NSObject PositionWithTimeIntervalSinceLatestBoot (double seconds);
```
Comment (rolfbjarne) [inline src/oslog.cs]: Method must contain a verb:

```suggestion
		NSObject GetPositionWithTimeIntervalSinceLatestBoot (double seconds);
```

And shouldn't the return type be OSLogPosition?

File: src/oslog.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		[NullAllowed, Export ("argumentStringValue")]
+		string ArgumentStringValue { get; }
+
+		[Export ("argumentUInt64Value")]
+		ulong ArgumentUInt64Value { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSObject))]
+	interface OSLogStore
```
Comment (spouliot) [inline src/oslog.cs]: considering the static factory methods it looks like this should be `[DisableDefaultCtor]`

File: src/oslog.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+using System;
+
+namespace OSLog {
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSObject))]
```
Comment (prmallic) [inline src/oslog.cs]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/oslog.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+		[Export ("formatString")]
+		string FormatString { get; }
+
+		[Abstract]
+		[Export ("subsystem")]
+		string Subsystem { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
```
Comment (prmallic) [inline src/oslog.cs]:
```suggestion
	[BaseType (typeof (OSLogEntry))]
```

File: src/oslog.cs [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	interface OSLogEntryActivity : OSLogEntryFromProcess
+	{
+		[Export ("parentActivityIdentifier")]
+		ulong ParentActivityIdentifier { get; }
+	}
+
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
```
Comment (prmallic) [inline src/oslog.cs]:
```suggestion
	[BaseType (typeof (OSLogEntry))]
```

File: src/oslog.cs [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+	interface OSLogEntryLog : OSLogEntryFromProcess, OSLogEntryWithPayload
+	{
+		[Export ("level")]
+		LogLevel Level { get; }
+	}
+
+
+	[Mac (10,15)]
+	[BaseType (typeof(OSLogEntry))]
```
Comment (prmallic) [inline src/oslog.cs]:
```suggestion
	[BaseType (typeof (OSLogEntry))]
```

File: src/oslog.cs [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+		[NullAllowed, Export ("argumentStringValue")]
+		string ArgumentStringValue { get; }
+
+		[Export ("argumentUInt64Value")]
+		ulong ArgumentUInt64Value { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSObject))]
```
Comment (prmallic) [inline src/oslog.cs]:
```suggestion
	[BaseType (typeof (NSObject))]
```

**Comment by wachaudh** [comment]
Closing because the Apple header is buggy. An issue has been filed and can be followed here: https://github.com/xamarin/xamarin-macios/issues/7450

**Pull request closed without merging**